configure.ac

```diff
@@ -507,6 +507,37 @@ AC_MSG_CHECKING([sould NCP support UART])
 AC_MSG_RESULT([${OPENTHREAD_ENABLE_UART}])
 
 #
+# Single Instance Optimization
+#
+
+AC_ARG_ENABLE(single-instance-opt,
```

Comment:
What is the use case for having a single instance but *without* the single instance optimization?  In the end, the more configuration options we have, the more configuration permutations we have to test.